mlir/lib/Dialect/Linalg/Transforms/Transforms.cpp

```diff
@@ -355,11 +360,18 @@ FailureOr<LowerPackResult> linalg::lowerPack(RewriterBase &rewriter,
 FailureOr<LowerUnPackOpResult>
 linalg::lowerUnPack(RewriterBase &rewriter, linalg::UnPackOp unPackOp,
                     bool lowerUnpadLikeWithExtractSlice) {
+  // TODO(issues/129004): Support MemRef UnPackOp. Temporarily return failure.
+  if (!unPackOp.hasPureTensorSemantics()) {
+    return failure();
+  }
+
   Location loc = unPackOp->getLoc();
   OpBuilder::InsertionGuard g(rewriter);
   rewriter.setInsertionPoint(unPackOp);
 
-  RankedTensorType packedTensorType = unPackOp.getSourceType();
+  // TODO: support non-ranked tensor types. ShapedType
+  RankedTensorType packedTensorType =
+      dyn_cast<RankedTensorType>(unPackOp.getSourceType());
```

Comment:
We are not going to support non-ranked types for pack/unpack ops. So please remove the TODO.

style nit: dyn_cast already spells the type, please use auto here. https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable